cache/config/config.go

```diff
@@ -0,0 +1,8 @@
+package config
+
+import "github.com/moby/buildkit/util/compression"
+
+type RefConfig struct {
+	Compression                   compression.Config
+	ConvertNonDistributableLayers bool
```

Comment:
`PreferNonDistributable` maybe?

Reply:
Sounds good. At the moment I have this the opposite tense because I wanted the zero value to leave the media types alone.

Reply:
Ah, I missed that. Zero value should be the previous behavior.

Storing the mediatype at all is a historic artifact that will probably go away in the future. We only want to save info about what the blob is: so compression and URLs for example. The user chooses the actual mediatype format as same blob can have many mediatypes.

That's why in another point I also recommended to just always use foreign mediatype when there are URLs in the descriptor and distributable when there are not. The original string doesn't really have much value as it depends on what order the same blob with different mediatypes was pulled.

Reply:
Fxied.